hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodiePartialUpdate.java

```diff
@@ -0,0 +1,114 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.common.model.HoodieKey;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.model.PartialUpdatePayload;
+import org.apache.hudi.common.testutils.HoodieTestDataGenerator;
+import org.apache.hudi.testutils.HoodieClientTestBase;
+import org.apache.spark.api.java.JavaRDD;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.hudi.common.testutils.HoodieTestDataGenerator.TRIP_SCHEMA;
+import static org.apache.hudi.common.testutils.HoodieTestDataGenerator.PARTIAL_TRIP_SCHEMA;
+import static org.apache.hudi.common.testutils.HoodieTestDataGenerator.AVRO_TRIP_SCHEMA;
+import static org.apache.hudi.common.testutils.Transformations.recordsToHoodieKeys;
+import static org.apache.hudi.common.util.ParquetUtils.readAvroRecords;
+import static org.apache.hudi.common.util.ParquetUtils.readAvroSchema;
+import static org.apache.hudi.common.util.ParquetUtils.readRowKeysFromParquet;
+import static org.apache.hudi.testutils.Assertions.assertNoWriteErrors;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestHoodiePartialUpdate extends HoodieClientTestBase {
+
+  @Test
+  public void testCopyOnWritePartialUpdate() {
+    final String testPartitionPath = "2020/12/28";
+    SparkRDDWriteClient client = getHoodieWriteClient(getPartialUpdateConfig());
+    dataGen = new HoodieTestDataGenerator(new String[] {testPartitionPath});
+
+    String commitTime1 = "001";
+    client.startCommitWithTime(commitTime1);
+
+    List<HoodieRecord> inserts1 =
+            dataGen.generatePartialUpdateInsertsStream(commitTime1, 100, false, TRIP_SCHEMA, false).collect(Collectors.toList()); // this writes ~500kb
+    List<HoodieKey> insertKeys1 = recordsToHoodieKeys(inserts1);
+    upsertAndCheck(client, insertKeys1, commitTime1, false);
+
+
+    String commitTime2 = "002";
+    client.startCommitWithTime(commitTime2);
+
+    List<HoodieRecord> inserts2 =
+        dataGen.generatePartialUpdateInsertsStream(commitTime2, 100, false, PARTIAL_TRIP_SCHEMA, false).collect(Collectors.toList()); // this writes ~500kb
+    List<HoodieKey> insertKeys2 = recordsToHoodieKeys(inserts2);
+    WriteStatus writeStatus = upsertAndCheck(client, insertKeys2, commitTime2, true);
```

Comment:
if users use this payload, and adding some fields in commits, what will happen? would you please add a test for this case?

Reply:
For MOR table, and any other changes to make MOR partial update works?

Reply:
MOR uses another PR, or continues to work here？
@leesf